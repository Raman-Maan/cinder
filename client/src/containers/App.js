import React, { Component } from 'react';
import logo from '../images/logo.svg';
import './App.css';

class App extends Component {
  state = {
    server_response: ''
  };

  getMessage = () => {
    return fetch('/api/user/example')
      .then(res => res.json())
      .catch(err => console.error(err));
  };

  componentDidMount() {
    this.getMessage()
      .then(msg => this.setState({ server_response: msg.example }))
      .catch(err => console.error(err));
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">{this.state.server_response}</p>
      </div>
    );
  }
}

export default App;
