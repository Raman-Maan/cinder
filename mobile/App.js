import React from 'react';
import Expo, { AppLoading } from 'expo';

import Home from './containers/home';

export default class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      fontLoaded: false,
    };
  }

  componentDidMount() {
    this.loadAssets();
  }

  async loadAssets() {
    await Expo.Font.loadAsync({
      Roboto: require('native-base/Fonts/Roboto.ttf'),
      Roboto_medium: require('native-base/Fonts/Roboto_medium.ttf'),
    });

    this.setState({ fontLoaded: true });
  }

  render() {
    if (!this.state.fontLoaded) {
      return <AppLoading />;
    }

    return (
      <Home />
    );
  }
}
