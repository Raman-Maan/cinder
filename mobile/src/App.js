import React from 'react';
import { View, StatusBar } from 'react-native';
import { StackNavigator } from 'react-navigation';

import Home from './containers/home';
import Profile from './containers/blank';
import Matches from './containers/matches';

const RootStack = StackNavigator({
  Home: {
    screen: Home,
  },
  Profile: {
    screen: Profile,
  },
  Matches: {
    screen: Matches,
  },
}, {
  initialRouteName: 'Matches',
  navigationOptions: {
    title: 'cinder',
  },
});

export default class App extends React.Component {
  render() {
    return (
      <RootStack />
    );
  }
}
