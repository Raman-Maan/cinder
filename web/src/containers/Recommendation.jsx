import React, { Component } from 'react';
import UserDetail from '../components/UserDetail';
import Auth from '../utils/authService';

import './styles/Recommendation.css';
export default class Recommendation extends Component {
  constructor(props) {
    super(props);
    this.fetchPotentialMatches = this.fetchPotentialMatches.bind(this);
    this.handleLike = this.handleLike.bind(this);
    this.handlePass = this.handlePass.bind(this);

    this.state = {
      potentialMatches: [],
      matchIndex: 0,
      currRecommendation: {}
    };
  }

  componentDidMount() {
    this.fetchPotentialMatches();
  }

  incrementPotentialMatchIndex() {
    this.setState(prevIndex => ({ matchIndex: prevIndex.matchIndex + 1 }));
    this.setState({
      currRecommendation: this.state.potentialMatches[this.state.matchIndex]
    });
  }

  handlePass() {
    this.submitUserAction('pass');
    this.incrementPotentialMatchIndex();
  }

  handleLike() {
    this.submitUserAction('like');
    this.incrementPotentialMatchIndex();
  }

  submitUserAction(userAction) {
    const matchedUser = this.state.potentialMatches[this.state.matchIndex].id;
    console.log(
      `Doing Action ${userAction} on ${this.state.currRecommendation.name}`
    );

    fetch(
      `/api/users/${Auth.loggedInUser.id}/matches/${matchedUser}/${userAction}`,
      { method: 'POST' }
    )
      .then(res => res.json())
      .then(res => {
        this.setState({
          // TODO
        });
      })
      .catch(err => console.error(err));
  }

  fetchPotentialMatches() {
    fetch(`/api/users/${Auth.loggedInUser.id}/recommendations`)
      .then(res => res.json())
      .then(res => {
        this.setState({
          potentialMatches: res.map(x => ({
            id: x.userID,
            name: x.userName,
            age: x.age,
            img: x.primaryPic,
            bio: x.userBio
          }))
        });
        this.setState({
          currRecommendation: this.state.potentialMatches[this.state.matchIndex]
        });
      })
      .catch(err => console.error(err));
  }

  render() {
    if (!this.state.potentialMatches.length) {
      return (
        <div className="empty">
          <h1 className="msg">There's no one new around you :(</h1>
        </div>
      );
    }

    const { img, name, age, bio } = this.state.currRecommendation;

    return (
      <UserDetail
        img={img}
        name={name}
        age={age}
        bio={bio}
        matchBtns={{
          handleLike: this.handleLike.bind(this),
          handlePass: this.handlePass.bind(this)
        }}
      />
    );
  }
}
