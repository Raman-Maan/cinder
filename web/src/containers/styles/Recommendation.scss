$bg-color: #ddd;
$empty-msg-color: #aaa;

.empty {
  display: flex;
  width: 100%;
  height: 100%;

  background-color: $bg-color;

  padding: 1em;

  justify-content: center;
  align-items: center;

  .msg {
    color: $empty-msg-color;
  }
}
