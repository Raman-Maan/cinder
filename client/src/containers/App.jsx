import React, { Component } from 'react';
import logo from '../assets/logo.svg';
import './App.css';

class App extends Component {
  static getMessage() {
    return fetch('/api/user/example')
      .then(res => res.json())
      .catch(err => console.error(err));
  }

  constructor() {
    super();
    this.state = {
      server_response: 'No request for message has been made'
    };
  }

  componentDidMount() {
    App.getMessage()
      .then(msg => this.setState({ server_response: msg.example }))
      .catch(err => console.error(err));
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">{this.state.server_response}</p>
      </div>
    );
  }
}

export default App;
