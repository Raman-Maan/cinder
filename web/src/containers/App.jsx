import React, { Component } from 'react';
import { Switch, Route, withRouter } from 'react-router-dom';
import PrivateRoute from '../components/PrivateRoute';
import './styles/App.css';
import { connect } from 'react-redux';

import Layout from './Layout';
import Login from './Login';
import Signup from './Signup';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Switch>
          <Route exact path="/login" component={Login} />
          <PrivateRoute
            exact
            path="/"
            authed={this.props.isAuthenticated}
            component={Layout}
          />
          <Route exact path="/signup" component={Signup} />
        </Switch>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated
});

export default withRouter(connect(mapStateToProps, null)(App));
