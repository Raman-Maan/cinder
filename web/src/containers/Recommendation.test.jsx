import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import fetchMock from 'fetch-mock';

import App from './App';
import Recommendation from './Recommendation';

describe('<Recommendation />', () => {
  const testPotentialMatches = [
    {
      userName: 'Some Name 1',
      userID: 1,
      age: 21,
      matchDate: '2018-02-03T18:09:05.000Z',
      primaryPic: null
    },
    {
      userName: 'Some Name 2',
      userID: 5,
      age: 22,
      matchDate: '2018-01-03T18:09:05.000Z',
      primaryPic: 'http://c.min.ms/m/b/13/13022/e8250eba.jpeg'
    },
    {
      userName: 'Some Name 3',
      userID: 3,
      age: 23,
      matchDate: '2018-02-06T18:09:05.000Z',
      primaryPic: null
    },
    {
      userName: 'Some Name 4',
      userID: 2,
      age: 24,
      matchDate: '2018-02-04T18:09:05.000Z',
      primaryPic: 'http://c.min.ms/m/b/13/13022/e8250eba.jpeg'
    }
  ];

  beforeEach(() => {
    fetchMock.reset();
    fetchMock.restore();
    fetchMock.get(/\/api\/users\/[0-9]*\/matches/, []);
    fetchMock.get(/\/api\/users\/[0-9]*\/recommendations/, testPotentialMatches);
  });

  afterAll(() => {
    fetchMock.reset();
    fetchMock.restore();
  });

  it('renders without crashing', () => {
    // simple smoke test
    shallow(<Recommendation />);
    mount(<Recommendation />);
  });

  it('should show the <Recommendation /> component by default', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['']}>
        <App />
      </MemoryRouter>
    );

    expect(wrapper.find(Recommendation)).toHaveLength(1);
  });

  it('renders with empty props correctly', () => {
    const component = renderer.create(<Recommendation />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
