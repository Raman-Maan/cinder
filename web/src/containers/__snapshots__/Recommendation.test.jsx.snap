// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Recommendation /> renders with empty props correctly 1`] = `
<div>
  <h1>
    There's no one new around you.
  </h1>
</div>
`;
